api/plumber.R

```diff
@@ -1,13 +1,80 @@
-library(tidyquant)
-library(tidyverse)
+#library(tidyquant)
+#library(tidyverse)
+library(glue)
+library(dplyr)
+library(pool)
+
+valid_tickers <- c(
+  Apple = "AAPL",
+  Amazon = "AMZN",
+  Facebook = "FB",
+  Google = "GOOG",
+  Intel = "INTC",
+  Microsoft = "MSFT"
+)
+
+db_path <- config::get("db", file = "config.yml")
+con <- pool::dbPool(drv = RSQLite::SQLite(), db = db_path)
+
+# in an online environment,
+# replace with 
+# tidyquant::tq_get(ticker, from = "2010-01-01")
+# to pull data directly from Yahoo
+# (note to remove the valid_tickers checks)
+get_price_data <- function(ticker, from = "2010-01-01", con) {
+  ticker <- glue::single_quote(ticker)
+  from <- as.integer(as.Date(from))
```

Comment:
This string -> date -> integer is not awesome ... I thought that the odbc driver handled the date conversion automatically? Is that not the case?

Reply:
Definitely agree. The SQLite driver is pretty lame wrt dates (SQLite does not have an inbuilt date type, so you need an integer), but I'll take a look if there is a nice way to clean up

Reply:
Sad... so we definitely want the ability to input a string, since the date type doesn't make a whole lot of sense in HTTP... so we have to convert to date, but it doesn't seem that the driver is smart enough to convert date -> integer to compare with the SQLite data... so I think we're kinda stuck here, unfortunately.